bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitImpl.java

```diff
@@ -44,45 +47,86 @@
+import io.github.hapjava.server.impl.crypto.HAPSetupCodeUtils;
 
 /**
  * Provides access to openHAB items via the HomeKit API
  *
  * @author Andy Lintner - Initial contribution
  */
-@Component(service = { Homekit.class }, configurationPid = "org.openhab.homekit", property = {
+@Component(service = { Homekit.class }, configurationPid = HomekitSettings.CONFIG_PID, property = {
         Constants.SERVICE_PID + "=org.openhab.homekit", "port:Integer=9123" })
 @ConfigurableService(category = "io", label = "HomeKit Integration", description_uri = "io:homekit")
 @NonNullByDefault
 public class HomekitImpl implements Homekit {
     private final Logger logger = LoggerFactory.getLogger(HomekitImpl.class);
-    private final NetworkAddressService networkAddressService;
-    private final HomekitChangeListener changeListener;
 
-    private HomekitSettings settings;
+    private final @NonNullByDefault({}) NetworkAddressService networkAddressService;
+    private final @NonNullByDefault({}) ConfigurationAdmin configAdmin;
+
+    private @NonNullByDefault({}) HomekitAuthInfoImpl authInfo;
+    private @NonNullByDefault({}) HomekitSettings settings;
     private @Nullable InetAddress networkInterface;
     private @Nullable HomekitServer homekitServer;
     private @Nullable HomekitRoot bridge;
-    private final HomekitAuthInfoImpl authInfo;
+    private @NonNullByDefault({}) HomekitChangeListener changeListener;
 
     private final ScheduledExecutorService scheduler = ThreadPoolManager
             .getScheduledPool(ThreadPoolManager.THREAD_POOL_NAME_COMMON);
 
     @Activate
     public HomekitImpl(@Reference StorageService storageService, @Reference ItemRegistry itemRegistry,
-            @Reference NetworkAddressService networkAddressService, Map<String, Object> config,
-            @Reference MetadataRegistry metadataRegistry) throws IOException, InvalidAlgorithmParameterException {
+            @Reference NetworkAddressService networkAddressService, @Reference MetadataRegistry metadataRegistry,
+            @Reference ConfigurationAdmin configAdmin, Map<String, Object> properties) {
         this.networkAddressService = networkAddressService;
-        this.settings = processConfig(config);
+        this.configAdmin = configAdmin;
+        this.settings = processConfig(properties);
         this.changeListener = new HomekitChangeListener(itemRegistry, settings, metadataRegistry, storageService);
-        authInfo = new HomekitAuthInfoImpl(storageService.getStorage(HomekitAuthInfoImpl.STORAGE_KEY), settings.pin);
-        startHomekitServer();
+        try {
+            authInfo = new HomekitAuthInfoImpl(storageService.getStorage(HomekitAuthInfoImpl.STORAGE_KEY), settings.pin,
+                    settings.setupId);
+            startHomekitServer();
+        } catch (IOException | InvalidAlgorithmParameterException e) {
+            logger.warn("Cannot activate HomeKit binding. {}", e.getMessage());
```

Comment:
```suggestion
            logger.warn("Cannot activate HomeKit binding. {}", e.getMessage());
            throw e;
```

If the constructor throws an exception, the framework should try to re-initialize.